src/data/nat/cast.lean

```diff
@@ -321,3 +321,18 @@ begin
 end
 
 end with_top
+
+namespace pi
+
+variables {α β : Type*}
+
+lemma nat_apply [has_zero β] [has_one β] [has_add β] :
+  ∀ (n : ℕ) (a : α), (n : α → β) a = n
+| 0     a := rfl
+| (n+1) a := show (n : α → β) a + (1 : α → β) a = n + 1, by { rw nat_apply, refl }
```

Comment:
I was hoping to be able to golf this to:
```lean
lemma nat_apply [semiring β] (n : ℕ) (a : α) : (n : α → β) a = n :=
(ring_hom.apply (λ _ : α, β) a).map_nat_cast n
```
But this has stronger assumptions (thanks to the lack of a `ring_without_mul_hom` I suppose), which I guess is not ideal.

Reply:
agreed